plugin.js

```diff
@@ -0,0 +1,20 @@
+const createServer = require('./server');
+
+let port;
+
+module.exports = function plugin(options) {
+  return function withConfig(nextConfig = {}) {
+    if (process.env.NODE_ENV !== 'production') {
+      if (!port) {
```

Comment:
This is great! 💯  I wonder if the config gets ran multiple times and that was the cause of the port already being used.